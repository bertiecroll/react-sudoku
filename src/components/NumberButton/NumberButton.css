.NumberButton {
  width: 2em;
  height: 2em;
  margin: .15em;
  line-height: 2em;
  font-size: 1.5em;
  text-align: center;
  border-radius: 5px;
  box-shadow: 1px 1px 1px 0px #000;
}

.NumberButton:nth-child(odd) {
  transform: rotate(5deg);
}

.NumberButton:nth-child(even) {
  transform: rotate(-5deg);
}

.NumberButton:hover {
  transform: scale(1.1);
  transition: transform .7s;
  cursor: pointer;
}

.not-available {
  background-color: grey;
  opacity: 0.2;
}

.not-available:hover {
  transform: none;
  cursor: not-allowed;
}
