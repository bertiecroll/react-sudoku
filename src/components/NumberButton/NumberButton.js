import React from 'react'
import PropTypes from 'prop-types'

import './NumberButton.css'

export default function NumberButton ({ number, isAvailable, onClickHandler }) {
  const classNames = `NumberButton ${isAvailable ? '' : 'not-available'}`

  return (
    <div
      className={classNames}
      onClick={onClickHandler}>
      {number}
    </div>
  )
}

NumberButton.defaultProps = {
  isAvailable: false,
  onClickHandler: () => {},
}

NumberButton.propTypes = {
  number: PropTypes.number.isRequired,
  isAvailable: PropTypes.bool.isRequired,
  onClickHandler: PropTypes.func.isRequired,
}
