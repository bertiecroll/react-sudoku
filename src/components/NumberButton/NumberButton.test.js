import React from 'react'
import { shallow } from 'enzyme'
import NumberButton from 'components/NumberButton'

describe('NumberButton', () => {
  it('renders div with when props.isAvailable is true', () => {
    const wrapper = shallow(<NumberButton number={1} isAvailable />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders div with not-available className when props.isAvailable is false', () => {
    const wrapper = shallow(<NumberButton number={1} />)
    expect(wrapper).toMatchSnapshot()
  })
})
