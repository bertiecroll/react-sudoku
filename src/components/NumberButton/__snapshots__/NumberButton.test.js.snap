// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberButton renders div with not-available className when props.isAvailable is false 1`] = `
<div
  className="NumberButton not-available"
  onClick={[Function]}
>
  1
</div>
`;

exports[`NumberButton renders div with when props.isAvailable is true 1`] = `
<div
  className="NumberButton "
  onClick={[Function]}
>
  1
</div>
`;
